libraries/lib-wave-track/WaveClip.h

```diff
@@ -184,6 +187,15 @@ class WAVE_TRACK_API WaveClip final :
 
    double GetPlayEndTime() const override;
 
+   /*!
+    * `GetPlayEndTime()` is the right edge of a left-closed and right-open
+    * interval. There is a little space (one track's sample period) left at the
+    * end of the clips, though. To get the exact location of the right border of
+    * this clip, use this method.
+    */
+   double GetBorderEndTime() const;
```

Comment:
This function is not yet used at the commit I am reviewing.  Can it be made private?

Reply:
Should instead `GetStretchRatio() / mRate` be the subtracted term?

Reply:
I had a need for it some place in `WaveTrack`, probably, then not anymore, I suppose since I added the new boundary-check methods. Will remove. Sorry for having wasted your time.